### PR TITLE
POD kNN

### PR DESCRIPTION
**Pull request by sslattery**
Adding kNN support to POD interface and fixing a bug in the POD iterator.

### CODE REVIEW

File: packages/Adapters/C_API/src/DTK_POD_PointCloudEntitySet.cpp [hunk 1]
```diff
@@ -113,15 +113,20 @@ EntityIterator POD_PointCloudEntitySet::entityIterator(
...
+    if ( 0 == topological_dimension )
+    {
+        iterator = POD_PointCloudEntityIterator( d_cloud_coords,
+                                                 d_global_ids,
+                                                 d_num_points,
+                                                 d_space_dim,
+                                                 d_layout,
+                                                 d_comm->getRank(),
+                                                 predicate );
+    }
```
Comment (dalg24) [inline packages/Adapters/C_API/src/DTK_POD_PointCloudEntitySet.cpp]: when would this would be called with `topological_dimension > 0`?

Comment (sslattery) [inline packages/Adapters/C_API/src/DTK_POD_PointCloudEntitySet.cpp]: it gets called in the default bounding box implementation for entity sets

File: packages/Adapters/C_API/test/tstPOD_C_API.cpp [hunk 2]
```diff
@@ -85,17 +81,61 @@ TEUCHOS_UNIT_TEST( C_API, create_apply_delete_map )
...
+  unsigned size = gold_data.size();
+  for (unsigned i = 0; i < size; ++i)
+  {
+    TEST_FLOATING_EQUALITY( gold_data[i], tgt_field[i], rel_tol );
+    TEST_COMPARE( std::abs(gold_data[i] - tgt_field[i]), <, abs_tol );
+  }
+}
+
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( C_API, knn_map )
```
Comment (dalg24) [inline packages/Adapters/C_API/test/tstPOD_C_API.cpp]: This is the same as the previous ``TEUCHOS_UNIT_TEST ``, ``options`` is the only thing that changes.
I'd advocate for a ``for (std::string const & options : { ... }) { ... }``

Comment (sslattery) [inline packages/Adapters/C_API/test/tstPOD_C_API.cpp]: i like having separate tests so its more obvious what is failing - in this case radius search vs kin

**Pull request merged**